qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,647 @@
+from typing import Any, Dict, Set, Tuple, Union, List, Optional
+
+from qiskit.circuit import Gate
+from qiskit import QuantumCircuit
+from qiskit.pulse import (
+    Schedule,
+    DriveChannel,
+    ControlChannel,
+    MeasureChannel,
+    Call,
+    Play,
+    ShiftPhase,
+    SetPhase,
+    ShiftFrequency,
+    SetFrequency,
+)
+from qiskit.pulse.channels import PulseChannel
+from qiskit.pulse.transforms import inline_subroutines
+from qiskit.circuit import Parameter, ParameterExpression
+from qiskit_experiments.calibration.exceptions import CalibrationError
+from qiskit_experiments.calibration.parameter_value import ParameterValue
+
+ParameterKey = namedtuple("ParameterKey", ["schedule", "parameter", "qubits"])
+
+
+class Calibrations:
+    """
+    A class to manage schedules with calibrated parameter values.
+    Schedules are stored in a dict and are intended to be fully parameterized,
+    including the index of the channels. The parameter values are stored in a
+    dict where parameters are keys. This class supports:
+    - having different schedules share parameters
+    - allows default schedules for qubits that can be overridden of specific qubits.
+
+    Parametric channel naming convention. Channels must be name according to a predefined
+    pattern so that self can resolve the channels and control channels when assigning
+    values to the parametric channel indices. This is pattern is "^ch\d[.\d]*\${0,1}[\d]*$",
+    examples of which include "ch0", "ch1", "ch0.1", "ch0$", "ch2$3", and "ch1.0.3$2".
+    The "." delimiter is used to specify the different qubits when looking for control
+    channels. The optional $ delimiter is used to specify which control channel to use
+    if several control channels work together on the same qubits. For example, if the
+    control channel configuration is {(3,2): [ControlChannel(3), ControlChannel(12)]}
+    then given qubits (2, 3) the name "ch1.0$1" will resolve to ControlChannel(12) while
+    "ch1.0$0" will resolve to ControlChannel(3).
+    """
+
+    def __init__(self, control_config: Dict[Tuple[int, ...], List[ControlChannel]] = None):
+        """
+        Initialize the instructions from a given backend.
+
+        Args:
+            control_config: A configuration dictionary of any control channels. The
+                keys are tuples of qubits and the values are a list of ControlChannels
+                that correspond to the qubits in the keys.
+        """
+
+        self._controls_config = control_config if control_config else {}
+
+        # Dict of the form: (schedule.name, parameter.name, qubits): Parameter
+        self._parameter_map = {}
+
+        # Reverse mapping of _parameter_map
+        self._parameter_map_r = defaultdict(set)
+
+        # Default dict of the form: (schedule.name, parameter.name, qubits): [ParameterValue, ...]
+        self._params = defaultdict(list)
+
+        self._schedules = {}
+
+        self._channel_pattern = "^ch\d[.\d]*\${0,1}[\d]*$"
```

Comment:
Is this class variable rather than instance variable?

Reply:
I think it could be. I'm a bit agnostic to class or instance variable here. Perhaps one advantage of a class variable is that it puts an emphasis in the code on this naming convention which is good. See here 0f60d1f.